chall08/gbudau.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+_VALID_ACTIONS = "IKNSWE"
+_INNER_GRID = "M0*"
+_OUTER_GRID = "+E"
+_WALL = "+"
+_MINE = "*"
+_ROBOT = "M"
+_EXIT = "E"
+_INIT = "I"
+_STOP = "K"
+
+def get_minefield():
+    minefield = []
+    err = False
+
+    try:
+        line = input()
+        while line:
+            minefield.append(line)
+            line = input()
+    except EOFError:
+        err = True
+        return minefield, err
+    return minefield, err
+
+def get_actions():
+    err = False
+    actions = ""
+
+    try:
+        actions = input()
+    except EOFError:
+        err = True
+        return actions, err
+    return actions, err
+
+def equal_width(minefield):
+    length = len(minefield[0])
+
+    for row in minefield:
```

Comment:
This could be: `return all([len(row) == len(minefield[0]) for row in minefield])` in 1 line.

Reply:
Good point, one adition is to drop the list parenthesis[] so it will short-circuit on the first time the condition is not meet instead of evaluating all rows.
`return all(len(row) == len(minefield[0]) for row in minefield)`

Reply:
Wow. Nice one 👍 Didn't think about it.